ethstats/ethstats.go

```diff
@@ -211,93 +253,215 @@ func (s *Service) loop() {
 				case txCh <- struct{}{}:
 				default:
 				}
-
 			// node stopped
 			case <-txSub.Err():
 				break HandleLoop
 			case <-headSub.Err():
 				break HandleLoop
+			case delegateSignMsg := <-istDelegateSignCh:
+				var statsPayload StatsPayload
+				err := json.Unmarshal(delegateSignMsg.Payload, &statsPayload)
+				if err == nil {
+					if s.backend.ProxyNode() != nil {
```

Comment:
both if statements seem to be doing the same thing, so again you could condense them into one

Reply:
Oops didn't realize they were actually different